### PR TITLE
api/test_api.py: Add /token endpoint test for negative path

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes https://github.com/kernelci/kernelci-api/issues/67

 Negative testing of /token endpoint by providing incorrect
password.

Signed-off-by: Jeny Sadadia <jeny.sadadia@gmail.com>

### CODE REVIEW

File: api/test_api.py [hunk 1]
```diff
@@ -96,3 +96,33 @@ def test_me_endpoint(mock_get_current_user):
     assert response.status_code == 200
     assert ('_id' and 'username' and 'hashed_password' and
             'active') in response.json()
+
+
+def test_token_endpoint_negative_path(mock_db_find_one):
```
Comment (mgalka) [inline api/test_api.py:101]: I'd rather choose more informative name. like `test_token_endpoint_wrong_password`, there can be more than one "negative" path to test in future. It also simplifies reading test logs when the test case names are more descriptive/

**Review by mgalka — CHANGES_REQUESTED** [review]

File: api/test_api.py (hunk 1 above)
Comment (JenySadadia) [reply api/test_api.py]: Renamed the function.

**Review by mgalka — APPROVED** [review]
